flow/record/adapter/splunk.py

```diff
@@ -51,38 +85,161 @@ def splunkify(record, tag=None):
+        self.protocol = next((protocol for protocol in SPLUNK_PROTOCOLS if protocol.value == url_scheme), None)
+        self.sourcetype = next((source for source in SPLUNK_SOURCETYPES if source.value == sourcetype), None)
+
+        if not self.sourcetype:
+            raise ValueError(f"Unsupported source type {sourcetype}.")
+        if not self.protocol:
+            raise ValueError(f"Unsupported protocol {url_scheme}.")
+        if self.protocol == SPLUNK_PROTOCOLS.TCP and self.sourcetype != SPLUNK_SOURCETYPES.RECORDS:
+            raise ValueError("For sending data to splunk over TCP, only the 'records' sourcetype is allowed.")
+
+        self.host = parsed_url.hostname
+        self.port = parsed_url.port
         self.tag = tag
-        self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM, socket.SOL_TCP)
-        self.sock.connect((host, port))
-        self.descriptors = {}
+        self.record_buffer = set()
         self._warned = False
+        self.packer = None
+
+        if self.protocol == SPLUNK_PROTOCOLS.TCP:
+            self.sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM, socket.SOL_TCP)
+            self.sock.connect((self.host, self.port))
+
+        elif self.protocol in [SPLUNK_PROTOCOLS.HTTP, SPLUNK_PROTOCOLS.HTTPS]:
+            if not HAS_REQUESTS:
+                raise ImportError("The requests library is required for sending data over HTTP(S).")
+            self.token = token
+
+            # Assume verify=True unless specified otherwise.
+            self.verify = not (str(ssl_verify).lower() in ("0", "false"))
+
+            scheme = self.protocol.value
+            endpoint = "event" if self.sourcetype != SPLUNK_SOURCETYPES.RECORDS else "raw"
+            port = f":{self.port}" if self.port else ""
+            self.url = f"{scheme}://{self.host}{port}/services/collector/{endpoint}?auto_extract_timestamp=true"
+            if not self.token:
+                raise ValueError("An authorization token is required for the HTTP collector.")
+            if not self.token.startswith("Splunk "):
+                self.token = "Splunk " + self.token
+            if not self.verify:
+                log.warning("Certification verification is disabled.")
+            self.headers = {
+                "Authorization": self.token,
+                "X-Splunk-Request-Channel": str(uuid.uuid4()),
+            }
+
+        if self.sourcetype == SPLUNK_SOURCETYPES.JSON:
+            self.packer = JsonRecordPacker(indent=4, pack_descriptors=False)
+
+    def _send_http(self, data: bytes, flush: bool = False) -> None:
+        if data:
+            self.record_buffer.add(data)
+        if len(self.record_buffer) < RECORD_BUFFER_LIMIT and not flush:
+            return
+        buf = b""
+        for record_data in self.record_buffer:
+            buf += record_data + b"\n"
+
+        if not len(buf):
+            return
+
+        # Remove the last newline as it is duplicate.
+        buf = buf[:-1]
```

Comment:
```suggestion
        buf = b"\n".join(self.record_buffer)

        if not len(buf):
            return
```

A set is unordered, not a big deal here but could be annoying when debugging.
Are you using a set to deduplicate records? if not i would just use a list.

Also, consider moving the buffering code outside this method.